go/utils/retry.go

```diff
@@ -4,11 +4,13 @@ import "time"
 
 // Retry will execute the given function until it returns true with the given
 // delay (milliseconds) between calls
-func Retry(delayInMilliseconds int, attemptFn func() bool) {
+func Retry(delayInMilliseconds int, attemptFn func(int) bool) {
+	initialDelay := delayInMilliseconds
 	for {
-		if attemptFn() {
+		if attemptFn(initialDelay) {
 			return
 		}
-		time.Sleep(time.Duration(delayInMilliseconds) * time.Millisecond)
+		initialDelay = initialDelay * 2
```

Comment:
Can we move this multiplication to after the sleep so that? Currently we never sleep the passed in amount (the first amount that is slept if double that value). This also sends in the next delay to the attemptFn so it won't need to double the passed in value